app/src/main/java/org/simple/clinic/sync/indicator/SyncIndicatorViewController.kt

```diff
@@ -52,11 +54,11 @@ class SyncIndicatorViewController @Inject constructor(
     val syncProgress = lastSyncedStateStream.filter { it.lastSyncProgress != null }
 
     val showSyncIndicatorState = Observables
-        .combineLatest(screenCreated, syncProgress)
-        { _, stateStream ->
+        .combineLatest(screenCreated, syncProgress, configProvider.toObservable())
+        { _, stateStream, config ->
           val indicatorState = when (stateStream.lastSyncProgress!!) {
-            SUCCESS -> syncIndicatorState(stateStream.lastSyncSuccessTimestamp)
-            FAILURE -> syncedFailureState(stateStream.lastSyncSuccessTimestamp)
+            SUCCESS -> syncIndicatorState(stateStream.lastSyncSuccessTimestamp!!, config.syncFailureThreshold)
+            FAILURE -> syncedFailureState(stateStream.lastSyncSuccessTimestamp, config.syncFailureThreshold)
```

Comment:
What do you think about renaming `lastSyncSuccessTimestamp` -> `lastSuccessfulSyncTime`? Or `lastSyncSucceededAt`? 

`lastSyncSuccessTimestamp` doesn't sound very natural.

Reply:
Renamed.